resources/tenant-onboarding-app.yaml

```diff
@@ -482,40 +492,30 @@ Resources:
               awslogs-region: !Ref AWS::Region
               awslogs-stream-prefix: ecs
           Environment:
-            Fn::If:
-              - ProvisionRDS
-              - - Name: AWS_REGION
-                  Value: !Ref AWS::Region
-                - Name: SAAS_BOOST_ENV
-                  Value: !Ref Environment
-                - Name: TENANT_ID
-                  Value: !Ref TenantId
-                - Name: SAAS_BOOST_RESOURCES_BUCKET
-                  Value: !Sub '{{resolve:ssm:/saas-boost/${Environment}/RESOURCES_BUCKET}}'
-                - Name: SAAS_BOOST_EVENT_BUS
-                  Value: !Ref EventBus
-                - Name: METRICS_STREAM
-                  Value: !Ref MetricsStream
-                - Name: DB_HOST
-                  Value: !GetAtt rds.Outputs.RdsEndpoint
-                - Name: DB_PORT
-                  Value: !Ref RDSPort
-                - Name: DB_NAME
-                  Value: !Ref RDSDatabase
-                - Name: DB_USER
-                  Value: !Ref RDSUsername
-              - - Name: AWS_REGION
-                  Value: !Ref AWS::Region
-                - Name: SAAS_BOOST_ENV
-                  Value: !Ref Environment
-                - Name: TENANT_ID
-                  Value: !Ref TenantId
-                - Name: SAAS_BOOST_RESOURCES_BUCKET
-                  Value: !Sub '{{resolve:ssm:/saas-boost/${Environment}/RESOURCES_BUCKET}}'
-                - Name: SAAS_BOOST_EVENT_BUS
-                  Value: !Ref EventBus
-                - Name: METRICS_STREAM
-                  Value: !Ref MetricsStream
+              - Name: AWS_REGION
+                Value: !Ref AWS::Region
+              - Name: SAAS_BOOST_ENV
+                Value: !Ref Environment
+              - Name: TENANT_ID
+                Value: !Ref TenantId
+              - Name: SAAS_BOOST_RESOURCES_BUCKET
+                Value: !Sub '{{resolve:ssm:/saas-boost/${Environment}/RESOURCES_BUCKET}}'
+              - Name: SAAS_BOOST_EVENT_BUS
+                Value: !Ref EventBus
+              - Name: METRICS_STREAM
+                Value: !Ref MetricsStream
+              - Name: DB_HOST
+                Value: !If [ProvisionRDS, !GetAtt rds.Outputs.RdsEndpoint, !Ref AWS::NoValue]
+              - Name: DB_PORT
+                Value: !If [ProvisionRDS, !Ref RDSPort, !Ref AWS::NoValue]
+              - Name: DB_NAME
+                Value: !If [ProvisionRDS, !Ref RDSDatabase, !Ref AWS::NoValue]
+              - Name: DB_USER
+                Value: !If [ProvisionRDS, !Ref RDSUsername, !Ref AWS::NoValue]
+              - Name: S3_BUCKET
+                Value: !If [UseS3, !Ref S3ExtensionBucket, !Ref AWS::NoValue]
+              - Name: S3_ROOT_PREFIX
+                Value: !If [UseS3, !Sub '${ServiceResourceName}/${TenantId}/', !Ref AWS::NoValue]
```

Comment:
I don't think we should include the tenant id as part of the prefix ENV variable. It doesn't match the existing S3 patterns we have and if we get to a place where we have pooled compute resources, the SaaS application will have to set the tenant id at runtime.